### PR TITLE
Add a simple draft-publish workflow for use with plates (SimpleResource), refs #1031

### PR DESCRIPTION
**Pull request by hackmastera**
It's nearly identical to the folder workflow; the only difference is the names of the states / transitions.

### CODE REVIEW

File: app/models/draft_publish_workflow.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    event :publish do
+      transitions from: :draft, to: :published
+    end
+    event :unpublish do
+      transitions from: :published, to: :draft
+    end
+  end
+
+  def suppressed?
+    false
```
Comment (escowles) [inline app/models/draft_publish_workflow.rb]: Shouldn't this check `draft?` ?

Comment (hackmastera) [reply app/models/draft_publish_workflow.rb]: Hm, I didn't really see this used so I had tried removing it but it clearly needs more attention.

In folder workflow, which I based this on, it's always false. Do you know how this works @escowles ?
https://github.com/pulibrary/figgy/blob/master/app/models/folder_workflow.rb#L25-L27

Comment (hackmastera) [reply app/models/draft_publish_workflow.rb]: here it seems to be an optional other state: https://github.com/pulibrary/figgy/blob/master/app/models/book_workflow.rb#L5

I'm not sure it's' used in indexing...

Comment (hackmastera) [reply app/models/draft_publish_workflow.rb]: For access we need something like this https://github.com/pulibrary/figgy/blob/master/app/indexers/ephemera_folder_indexer.rb#L19

Comment (hackmastera) [reply app/models/draft_publish_workflow.rb]: I added the false back in with a comment that I think is accurate. Also created #1040.

Comment (escowles) [reply app/models/draft_publish_workflow.rb]: I'm not sure if the `suppressed?` method is really being used for anything, so maybe we can keep as-is, and use #1040 to explore whether we need it, or if there is some refactoring to do between indexers and `Ability` to correctly suppress unpublished items?

Comment (hackmastera) [reply app/models/draft_publish_workflow.rb]: sounds good. I'll add a note to #1040

**Comment by hackmastera** [comment]
This is ready for review; I think we'll need it in before we merge the new model.

**Pull request merged**